application/common/models/Method.php

```diff
@@ -25,10 +25,33 @@ public function rules()
                 ],
 
                 [
+                    ['verified', 'verification_attempts'], 'default', 'value' => 0,
+                ],
+
+                [
+                    'verification_code', 'default', 'when' => function() { return $this->getIsNewRecord(); },
```

Comment:
I believe in Yii2 the 'on' attribute must reference a defined scenario: http://www.yiiframework.com/doc-2.0/guide-structure-models.html#scenarios

So I don't think we can just do on insert or on update.